### ISSUE TITLE
Why isn't the API chainable?

### ISSUE THREAD

**Issue by louisremi** [paperjs/paper.js#605]
I was going to name this bug "Make the API chainable", but I just read this comment in `Rectangle.setCenter` source code: `// A special setter where we allow chaining, because it comes in handy in a couple of places in core.`

So it seems you deliberately chose to not to allow chaining, which puzzles me... Are there such cases where it isn't handy? Even more surprising, parts of the API are actually chainable: this is the case for `Matrix` methods for instance.

I am strongly in favor of making the API constantly chainable. This is a characteristic of jQuery that made it so popular: it reduces verbosity and thus improves readability.

**Comment by louisremi** [comment]
> For example, I find it really counter-intuitive that `$('body').append('<div />')` returns a reference to `<body>` rather than the newly created `<div />`.

This is consistent with the rest of jQuery's API, and you can use `$('<div />').appendTo('body')` if you need to chain on the div. We will probably have to agree that we disagree because I find `(new paper.Layer()).addChild(new paper.Path())` returning the path to be very counter-intuitive. Paper.js could have a similar `(new paper.Path(pathData)).addTo(layer);`, that wouldn't hurt.

Moving forward, I think all methods that currently don't return anything could be made chainable. I actually only discovered that the API wasn't chainable when I used `Path.prototype.reverse` and `Path.prototype.simplify` for the first time.

**Comment by lehni** [comment]
Yes, I agree that methods not returning things currently should allow chainability. Will spend some time going through the API soon to implement this. Anything beyond thatwould break the API .

And I like the proposed #addTo(), it's short too. Could also work on a project, for layers?

**Comment by arctwelve** [comment]
> I am strongly in favor of making the API constantly chainable. This is a characteristic of jQuery that made it so popular: it reduces verbosity and thus improves readability.

It's what made jQuery the source of some of the most miserable code ever written. Chaining is a sloppy idiom of JavaScript. Lines ending in dots, instead of semi-colons, make code much harder to read. Please don't add it to paper.js

**Comment by lehni** [comment]
Yes that was precisely my feeling also. And I am weary of ending up with tons of `return this` statements in the source-code that are impossible to minify further, so I haven't done anything here yet. I think my answer in https://github.com/paperjs/paper.js/issues/605#issuecomment-68107223 is summing it up nicely.

**Comment by megawac** [comment]
> that are impossible to minify further

I wouldn't worry about that concern, gzip will nearly completely handle this case.

I think APIs such as path (as in my example) should be chainable. With jQuery the main issue is some of the APIs return a jquery object while others return DOM objects/other
